### PR TITLE
Include runtime dependencies for taggers in `gcr.io/k8s-skaffold/skaffold`

### LINKED ISSUE #1979: skaffold docker doesn't tag correctly
When I run `skaffold run` on CircleCI with the image of `gcr.io/k8s-skaffold/skaffold`, following error occurs.

* Case 1
For the following,
```
  tagPolicy:
    gitCommit: {}
```
Error is
```
WARN[0000] Unable to find git commit: starting command &{git [git describe --tags --always] [] ../ <nil> 0xc0000f6720 0xc0000f6730 [] <nil> <nil> <nil> <nil> 0xc000809940 false [] [0xc0000f6720 0xc0000f6730] [0xc0000f6718 0xc0000f6728] [] <nil> <nil>}: exec: "git": executable file not found in $PATH 
```

* Case 2
For the dateTime tagging,
```
  tagPolicy:
    dateTime:
      timezone: "Asia/Tokyo"
```
Error is 
```
Generating tags...
 - gcr.io/{project}/{imagename} -> FATA[0000] generating tag: generating tag for gcr.io/{project}/{imagename}: bad timezone provided: "Asia/Tokyo", error: open /usr/local/go/lib/time/zoneinfo.zip: no such file or directory 
Exited with code 1
```
I guess the cause is required files/command are not in built docker image.
https://github.com/GoogleContainerTools/skaffold/blob/master/deploy/skaffold/Dockerfile

### Expected behavior
tagPolicy: gitCommit / dateTime should work correctly even if using docker image

### Actual behavior
- gitCommit: WARN error and tag is always `dirty`
- dateTime: FATAL error and exit

### Information

- Skaffold version: v0.27.0
- Operating system: gcr.io/k8s-skaffold/skaffold
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta7
kind: Config
build:
  artifacts:
    - image: gcr.io/{project}/{imagename}
      context: ../
  tagPolicy:
    gitCommit: {}
deploy:
  kubectl:
    manifests:
      - k8s/staging/configmap_nginx_vhost-stg.yml
      - k8s/staging/deployment-stg.yml
profiles:
  - name: production
    build:
      tagPolicy:
        dateTime:
          timezone: "Asia/Tokyo"
    deploy:
      kubectl:
        manifests:
          - k8s/production/*
  - name: staging
    build:
      tagPolicy:
        dateTime:
          timezone: "Asia/Tokyo"
    deploy:
      kubectl:
        manifests:
          - k8s/staging/*
```

### Steps to reproduce the behavior

1. put skaffold file in current directory.
2. `docker run -ti --rm -v $PWD:/mnt  gcr.io/k8s-skaffold/skaffold  /bin/bash -c 'cd /mnt/ && skaffold run -p staging'`

### PR DESCRIPTION
**Pull request by corneliusweig**
Also extend integration tests to assert on the expected image tags.
In addition, remove `helm-${HELM_VERSION}-linux-amd64.tar.gz` from the skaffold image.

Fixes #1979

### CODE REVIEW

**Comment by balopat** [comment]
can you please rebase this to master? the kokoro failure is fixed now there. Thank you!

File: deploy/skaffold/Dockerfile [hunk 1]
```diff
@@ -14,8 +14,12 @@
 
 FROM gcr.io/gcp-runtimes/ubuntu_16_0_4 as runtime_deps
 
+# needed for dateTime tagger
+COPY --from=golang:1.11 /usr/local/go/lib/time/zoneinfo.zip /usr/local/go/lib/time/zoneinfo.zip
```
Comment (balopat) [inline deploy/skaffold/Dockerfile]: hmmm...this smells a little bit for me...and not just me: https://github.com/golang/go/issues/21881#issuecomment-370125690 - maybe we should try 4d63.com/tz ?

Comment (corneliusweig) [reply deploy/skaffold/Dockerfile]: Yes, of course! With `4d63.com/tz` the dateTime tagger will also work on windows :)

**Comment by corneliusweig** [comment]
@balopat PTAL

**Comment by dgageot** [comment]
@corneliusweig kokoro tests are failing:

```
--- FAIL: TestBuild (6.96s)
    --- PASS: TestBuild/docker_build (6.60s)
    --- FAIL: TestBuild/git_tagger (0.01s)
        build_test.go:160: exit status 128
```

**Comment by corneliusweig** [comment]
...that was hard to find :sweat:

**Pull request merged**